controller/linodecluster_controller.go

```diff
@@ -157,6 +157,11 @@ func (r *LinodeClusterReconciler) reconcile(
 		return res, err
 	}
 
+	if err := clusterScope.SetCredentialRefTokenForLinodeClients(ctx); err != nil {
```

Comment:
I think we should set before we handle the deletes. When object(LinodeCluster in this case) is marked for deletion, it will enter the `if !clusterScope.LinodeCluster.DeletionTimestamp.IsZero() {` block and never reach this